### ISSUE TITLE
Training and test set mixed up

### ISSUE THREAD

**Issue by Filco306** [adityagaydhani14/Toxic-Language-Detection-in-Online-Content#1]
Dear Adtiya, 

First of all, thank you for a very interesting paper and approach! I really liked it all and found it very interesting to read. I even decided to use your paper as a baseline for my own project, where I decide to use your work as a comparison, and run my tests on the exact same data. 

However, I have discovered a problem with your work. This is actually quite a big problem, and it alters your results significantly. In your experiments, you use upsampling to compensate for imbalance, which is a reasonable approach. However, this is done before splitting test and train set. Thus, two big faults occurs. 

1. The training set contains duplicates. 
2. Not only does it contain duplicates, but it also contains tweets that are also in the training set. 

Removing all duplicates and all tweets existing in the training set these removes about 79 % of the whole test set. This shows that the test set is highly biased, and I am afraid this leads to your results being invalid. I did rerun your code with correct test and training sets, and although it still has a good accuracy (~86%), it is not near what you claim. Quoting you in my work, I simply cannot reference your accuracy that you claim. However, I would be happy if you were to disprove me, and the results were to be as good as you claim. 

Please let me know how you choose to follow up on this. If you write me your email, I can send you some proof of it all. I hope this has been of help, and let me know if there is anything else you would like to know. 

Best,
Filip

**Comment by vebits** [comment]
I did the same checking as @Filco306 and found that around 74% of the test data is either duplicate or in the training data. I would also appreciate a comment on this as your claimed results are some of the best in the field.

**Comment by adityagaydhani14** [comment]
Hi  Vebjørn,

Thank you for showing interest in my paper! I understand your concern about
the data present on my GitHub account. I will suggest you to not use this
data for your research because it is not the same that have been used for
the paper. Only some parts of the code present in my GitHub repository have
been used for the research.

Moreover, the data that I used for the paper was a combination of three
other data sets, which are mentioned in the paper. The problem now is that
the exact data that I used for the paper cannot be reproduced because
Twitter has deleted most of the hateful and offensive tweets present in the
data. And the pre-processing stage requires you to get the tweet content
from the tweet IDs present in the original data set. This content cannot be
obtained anymore because the tweets have been deleted. So the data that is
present on GitHub was generated a long time after the paper was published
and its not the same as the one used in the paper.

According to my understanding, the results in the paper are unbiased and
are achieved by using proper data. If you have any further questions,
please let me know. I hope that you understand. On a side note, there are
many other good data sets on toxic language that have been published since
2018. I would suggest you to use some of those for your research. Wish you
good luck!

Best,
Aditya
